quality/integration/src/test/java/test/integration/blob/CasBlobStoreTest.java

```diff
@@ -339,4 +422,95 @@ public void testListTables() {
             assertNotEquals(tableIter.next().getName(), TABLE);
         }
     }
-}
\ No newline at end of file
+
+    @Test
+    public void testScanMetadata() throws Exception {
+        String blobId1 = "1";
+        String blobId2 = "2";
+
+        assertEquals(Iterators.size(_store.scanMetadata(TABLE, null, Long.MAX_VALUE)), 0);
+
+        verifyNotExists(blobId1);
+        verifyNotExists(blobId2);
+        Date now = new Date();
+
+        ImmutableMap<String, String> attributes1 = ImmutableMap.of("encoding", "image/jpeg", "name", "mycat.jpg", "owner", "clover");
+        byte[] blobData1 = randomBytes(0x812345);
+        verifyPutAndGet(blobId1, blobData1, attributes1);
+
+        ImmutableMap<String, String> attributes2 = ImmutableMap.of("encoding", "image/png", "name", "mycat2.png");
+        byte[] blobData2 = randomBytes(0x4321);
+        verifyPutAndGet(blobId2, blobData2, attributes2);
+
+        BlobMetadata blobMetadata1 = getBlobMetadata(blobId1, blobData1, now, attributes1);
+        BlobMetadata blobMetadata2 = getBlobMetadata(blobId2, blobData2, now, attributes2);
+
+        assertEqualsBlobMetadata(sortIterator(_store.scanMetadata(TABLE, null, Long.MAX_VALUE)), Lists.newArrayList(blobMetadata1, blobMetadata2).iterator());
+        assertEquals(Iterators.size(_store.scanMetadata(TABLE, null, 1)), 1);
+    }
+
+    private static Iterator<BlobMetadata> sortIterator(final Iterator<BlobMetadata> iterator) {
+        return StreamSupport
+                .stream(Spliterators
+                        .spliteratorUnknownSize(iterator, Spliterator.ORDERED), false)
+                .sorted(Comparator.comparing(BlobMetadata::getId))
+                .iterator();
+    }
+
+    @Test
+    public void testScanMetadataFromBlobIdExclusive() throws Exception {
+        String blobId1 = "1";
+
+        verifyNotExists(blobId1);
+        Date now = new Date();
+
+        ImmutableMap<String, String> attributes1 = ImmutableMap.of("encoding", "image/jpeg", "name", "mycat.jpg", "owner", "clover");
+        byte[] blobData1 = randomBytes(0x812345);
```

Comment:
minor: as we are only dealing with 1 blob in this scan test, rename all the vars without "1".

Reply:
will fix